src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleasesController.cs

```diff
@@ -190,12 +198,38 @@ public async Task<ActionResult<FileInfo>> UpdateChartFileAsync(Guid releaseId, G
         [RequestSizeLimit(int.MaxValue)]
         [RequestFormLimits(ValueLengthLimit = int.MaxValue, MultipartBodyLengthLimit = int.MaxValue)]
         public async Task<ActionResult<DataFileInfo>> AddDataZipFileAsync(Guid releaseId,
-            [FromQuery(Name = "name"), Required] string name, IFormFile zipFile)
+            [FromQuery(Name = "replacingFileId")] Guid? replacingFileId,
+            [FromQuery(Name = "name")] string subjectName,
+            IFormFile zipFile)
         {
             var user = await _userManager.GetUserAsync(User);
 
             return await _releaseFilesService
-                .UploadDataFilesAsZipAsync(releaseId, zipFile, name, user.Email)
+                .UploadDataFilesAsZip(releaseId: releaseId,
+                    zipFile: zipFile,
+                    userName: user.Email,
+                    replacingFileId: replacingFileId,
+                    subjectName: subjectName)
+                .HandleFailuresOrOk();
+        }
+
+        [HttpPost("release/{releaseId}/zip-data/replacing/{replacingId}")]
+        [Produces("application/json")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(404)]
+        [RequestSizeLimit(int.MaxValue)]
+        [RequestFormLimits(ValueLengthLimit = int.MaxValue, MultipartBodyLengthLimit = int.MaxValue)]
+        public async Task<ActionResult<DataFileInfo>> AddReplacementDataZipFileAsync(Guid releaseId,
```

Comment:
Shouldn't need this endpoint anymore right?

Reply:
Oops, thanks :octocat: